recipes-connectivity/iot-companion/iot-companion/iot-companion.sh

```diff
@@ -0,0 +1,652 @@
+------------------------------------------------------------------------------------ 
+MODULE:             $(basename "${BASH_SOURCE[0]}") rauc            
+                    local RAUC operation 
+------------------------------------------------------------------------------------  
+OPTIONS:
+    compat        - show system compatible string
+    status        - show rauc system status
+    detail        - show detailed rauc system status including bundle hashes
+    switch        - activate the mirror system to be active after next boot
+    noswitch      - activate the current system to be active after next boot
+
+    markbad       - mark this slot as BAD - after next boot rauc might fallback
+    marbad-other  - mark the mirror slot as BAD - after next boot rauc should boot current system
+
+    markgood       - mark this slot as GOOD
+    markgood-other - mark mirror slot as GOOD
+
+    config        - show the config file $RaucConfig
+    config-edit   - DANGEROUS: opens the config in vi
+    config-reload - Restarts the rauc systemd service to reload the config (this is needed after compatible string change)
+    
+EOF
+}
+
+usage_tunnel(){
+usage_submodule_header
+cat<<EOF
+  _                          _ 
+ | |                        | |
+ | |_ _   _ _ __  _ __   ___| |
+ | __| | | | '_ \| '_ \ / _ \ |
+ | |_| |_| | | | | | | |  __/ |
+  \__|\__,_|_| |_|_| |_|\___|_|
+------------------------------------------------------------------------------------ 
+MODULE:             $(basename "${BASH_SOURCE[0]}") tunnel            
+                    Remote Manager - SSH Tunnel through managed jump host
+------------------------------------------------------------------------------------ 
+OPTIONS:
+    start         - starts the service   # Tunnel can only be actovated if configured in backend!
+    stop          - stops the service
+    restart       - restarts the service
+    status        - systemd status for the service
+
+    log           - show the latest log messages
+
+    config        - show the config file $RemoteManagerConfig
+    config-edit   - DANGEROUS: opens the config in vi
+    
+    alive          - show active ssh connections
+    keydir         - Key storage location  
+    pubkey         - show the public key
+    
+EOF
+}
+
+
+usage() {
+  usage_header
+  usage_aws
+  echo
+  usage_update
+  echo
+  usage_rauc
+  echo
+  usage_tunnel
+  exit
+}
+
+cleanup() {
+  trap - SIGINT SIGTERM ERR EXIT
+  # script cleanup here
+}
+
+setup_colors() {
+  if [[ -t 2 ]] && [[ -z "${NO_COLOR-}" ]] && [[ "${TERM-}" != "dumb" ]]; then
+    NOFORMAT='\033[0m' RED='\033[0;31m' GREEN='\033[0;32m' ORANGE='\033[0;33m' BLUE='\033[0;34m' PURPLE='\033[0;35m' CYAN='\033[0;36m' YELLOW='\033[1;33m'
+  else
+    NOFORMAT='' RED='' GREEN='' ORANGE='' BLUE='' PURPLE='' CYAN='' YELLOW=''
+  fi
+}
+
+msg() {
+  echo >&2 -e "${1-}"
+}
+
+die() {
+  local msg=$1
+  local code=${2-1} # default exit status 1
+  msg "$msg"
+  exit "$code"
+}
```

Comment:
Instead of defining a custom function for printing an error and exiting, simply print the error message with `printf` and exit with an appropriate error code directly, when needed, e.g.:
```
printf "ERROR: Unknown argument foobar\n"
exit 1
```

Reply:
Replaced the custom function